src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,20 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new UserDataInvalidException("The user provided is null");
+        }
+
+        if (checkDataValidity(user)) {
+            return storageDao.add(user);
+        }
+        throw new UserDataInvalidException("The user data is not valid");
+    }
+
+    private boolean checkDataValidity(User user) {
+        return user.getLogin().length() >= 6
+                && user.getPassword().length() >= 6
```

Comment:
Move the hardcoded value for the minimum password length to a constant field with an informative name. This will make it easier to change the value in the future and makes the code more readable.